modules/branch-keystore-node/src/branch_keystore.ts

```diff
@@ -0,0 +1,693 @@
+    //# Because the storage interface can be a custom implementation the key store needs to verify correctness.
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# GetActiveBranchKey MUST verify that the returned EncryptedHierarchicalKey MUST have the requested `branch-key-id`.
+    needs(
+      activeEncryptedBranchKey.branchKeyId == branchKeyId,
+      'Unexpected branch key id.'
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# GetActiveBranchKey MUST verify that the returned EncryptedHierarchicalKey is an ActiveHierarchicalSymmetricVersion.
+    needs(
+      activeEncryptedBranchKey.type instanceof
+        ActiveHierarchicalSymmetricVersion,
+      'Unexpected type. Not an version record.'
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# GetActiveBranchKey MUST verify that the returned EncryptedHierarchicalKey MUST have a logical table name equal to the configured logical table name.
+    needs(
+      activeEncryptedBranchKey.encryptionContext[TABLE_FIELD] ==
+        this.logicalKeyStoreName,
+      'Unexpected logical table name.'
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# If the branch key fails to decrypt, GetActiveBranchKey MUST fail.
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# The operation MUST decrypt the EncryptedHierarchicalKey according to the [AWS KMS Branch Key Decryption](#aws-kms-branch-key-decryption) section.
+    const branchKey = await decryptBranchKey(this, activeEncryptedBranchKey)
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# This GetActiveBranchKey MUST construct [branch key materials](./structures.md#branch-key-materials)
+    //# according to [Branch Key Materials From Authenticated Encryption Context](#branch-key-materials-from-authenticated-encryption-context).
+    const branchKeyMaterials = constructBranchKeyMaterials(
+      branchKey,
+      activeEncryptedBranchKey
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getactivebranchkey
+    //# This operation MUST return the constructed [branch key materials](./structures.md#branch-key-materials).
+    return branchKeyMaterials
+  }
+
+  async getBranchKeyVersion(
+    branchKeyId: string,
+    branchKeyVersion: string
+  ): Promise<NodeBranchKeyMaterial> {
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# On invocation, the caller:
+    //#
+    //# - MUST supply a `branch-key-id`
+    //# - MUST supply a `branchKeyVersion`
+    needs(
+      branchKeyId && typeof branchKeyId === 'string',
+      'MUST supply a string branch key id'
+    )
+    needs(
+      branchKeyId && branchKeyVersion,
```

Comment:
Sorry, it's been a while since I looked at TS.

Does this ensure the BKV is a string, or only that it has been set?